### ISSUE TITLE
[Bug] Blackhole: ttnn.reshape of a ROW_MAJOR tensor to [N, 1] hard-deadlocks the device

### ISSUE THREAD

**Issue by kinginu** [tenstorrent/tt-metal#50191]
*Suggested labels: `blackhole`, `bug`, `data_movement`*

## Summary

On Blackhole, reshaping a large **ROW_MAJOR** `bfloat16` tensor to a column shape `[N, 1]`
(`N ≈ 250M`) on **interleaved DRAM** intermittently **hard-deadlocks the device**:
`ttnn.synchronize_device` never returns, **no exception is raised**, and only `SIGKILL` / `tt-smi -r`
recovers. It is **not** OOM — a genuine over-allocation raises a clean `TT_FATAL: Out of Memory`.

Root cause: reshaping to `[..., 1]` gives `dest_page_size_bytes = 1 * sizeof(bf16) = 2` bytes, far below
the Blackhole DRAM alignment of **64 bytes**. The RM reshape kernel then drains each source page into
2-byte destination pages, executing a `noc_async_write_barrier()` on **every** tiny write (~250M
barriered 2-byte interleaved-DRAM writes); a write completion eventually never returns and the barrier
waits forever.

## Environment

- Reproduced on **v0.70.0rc6**; the code path is **unchanged on `main` (v0.75.0-dev)**.
- Blackhole, single chip (`MeshDevice`), **interleaved DRAM**. Firmware 19.7.0, KMD 2.7.0.

## Reproduce

```python
import torch, ttnn
dev = ttnn.open_device(device_id=0)
T, G1 = 2500, 100001                       # rank [2500, 100001] bf16 TILE; N = T*G1 ≈ 250M
rank = ttnn.from_torch(torch.zeros(T, G1, dtype=torch.bfloat16),
                       dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=dev)
for i in range(60):
    tl = ttnn.to_layout(rank, ttnn.ROW_MAJOR_LAYOUT)
    rf = ttnn.reshape(tl, (T * G1, 1))     # [N,1] RM: dest_page_size = 2 bytes
    ttnn.synchronize_device(dev)           # <-- HANGS here at a nondeterministic iter
    ttnn.deallocate(rf); ttnn.deallocate(tl)
```

Reshaping to `[1, N]` instead never hangs. Intermittent — hangs at a **nondeterministic iteration**
(observed 8 / 14 / 24 / 43); `leak` / `scatter` / a sync between the two ops are not required — only the
`[N, 1]` (unit-last-dim) reshape is.

## Root cause

`ttnn/cpp/ttnn/operations/data_movement/reshape_view/device/device/rm_reshape_interleaved.cpp`, the
non-`can_be_clean` path (taken because `dest_page_size_bytes % 16 != 0`):

```cpp
uint32_t writable = dest_page_size_bytes - write_start_offset;   // = 2 for [N,1]
while (readable > 0) {
    noc_async_write_barrier();     // line 112: executed once per <=2-byte write, ~250M total
    ...
}
```

`reshape_rm_program_factory.cpp:45` computes `dest_page_size_bytes = output_log_shape[-1] * data_size`
with **no guard** for `dest_page_size_bytes < DRAM_ALIGNMENT` (64 B on Blackhole). The in-loop barrier
was introduced in #15572 (2024-12) and is unchanged since.

## Notes

No existing issue matches this path (surveyed); the closest, #40612, is a different
Wormhole / TILE-layout root cause. I've root-caused this and am happy to work on a fix/PR.
